src/legacy/core_plugins/kibana/public/dashboard/panel/panel_utils.ts

```diff
@@ -17,25 +17,38 @@
  * under the License.
  */
 
-import _ from 'lodash';
 import { i18n } from '@kbn/i18n';
-import { DEFAULT_PANEL_WIDTH, DEFAULT_PANEL_HEIGHT } from '../dashboard_constants';
+import _ from 'lodash';
 import chrome from 'ui/chrome';
+import { DEFAULT_PANEL_HEIGHT, DEFAULT_PANEL_WIDTH } from '../dashboard_constants';
+import { GridData, PanelId, PanelState } from '../selectors';
 
 const PANEL_HEIGHT_SCALE_FACTOR = 5;
 const PANEL_HEIGHT_SCALE_FACTOR_WITH_MARGINS = 4;
 const PANEL_WIDTH_SCALE_FACTOR = 4;
 
 export class PanelUtils {
-
   // 6.1 switched from gridster to react grid. React grid uses different variables for tracking layout
-  static convertPanelDataPre_6_1(panel) { // eslint-disable-line camelcase
+  // Earlier versions allowed panelIndex to be a number, explicity casting panelIndex to type string for compatibility
+  public static convertPanelDataPre_6_1(panel: {
+    panelIndex: PanelId;
```

Comment:
Since we learned that in the older versions `panelIndex` could be a number, this type annotation can also reflect this, i.e. `panelIndex: PanelId | number`.

Reply:
O! That makes a lot of sense but it's producing problems when I add it when I also include Partial<PanelState> as the return type for the function. I think it's grumpy with number since it conflicts with the designated type with PanelState in the types.ts

```
Type '{ panelIndex: string | number; gridData: GridData; col: number; row: number; size_x: number; size_y: number; version: string; }' is not assignable to type 'Partial<PanelState>'.
  Types of property 'panelIndex' are incompatible.
    Type 'string | number' is not assignable to type 'string | undefined'.
      Type 'number' is not assignable to type 'string | undefined'.
```

I'm on the fence on how needed it is to include the `panelIndex: PanelId | number` (pro of extra type checking if it results in not having the return type of Partial<PanelState>) - but I might totally be off base on that and am open to your thoughts on it?

Reply:
If we err on the side of having our type annotations being reflective of what values are actually being passed around, this can help with documentation and future maintainability.  So if updating this leads to other errors... those errors give us a clue as to something else that is currently incompatible with the actuality of what the potential values should be.  In this case, since this change leads to an error in the expected return type, one possible change is the re-assign the value of `panelIndex` so that it is a `string`, i.e. `panel.panelIndex = panel.panelIndex.toString()` (as it appears in another file as well, I think.)

Reply:
Ok I was able to resolve the errors mentioned above and keep the Partial<PanelState> by going to the types.ts file `src/legacy/core_plugins/kibana/public/dashboard/selectors/types.ts` and changing the line `export type PanelId = string;` to `export type PanelId = string | number;` This resolved the problems, because PanelId was only permitted to be type string and mucking up the Partial<PanelState> type. 

I'm thinking this makes sense to change the types.ts file to account for PanelId to be either string or number. But wanted to confirm my logic and make sure that this doesn't make `panelIndex: PanelId | number` in panel_utils redundant. I don't think it's too redundant since it just brings attention to what we were mentioning earlier that panelIndex may be string or number.

Reply:
Running d420142 to confirm that the other changes were fine and rolling back my implementation.